### PR TITLE
[TS] LPS-163767 Make sure to return a valid defaultLanguaId

### PR DESCRIPTION
**Pull request by tototrinh**
Hi guys, please help review this PR and leave your comments.
Ticket: https://issues.liferay.com/browse/LPS-163767
Previous PR:
https://github.com/liferay-forms/liferay-portal/pull/1958
cc @rodrigopaulino

### CODE REVIEW

**Comment by tototrinh** [comment]
ci:test:sf

**Comment by tototrinh** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js [hunk 1]
```diff
@@ -41,6 +41,12 @@ export default function LocalesList({
 	const sortedLocales = useMemo(() => {
 		const availableLocalesArray = Array.from(availableLocales.values());
 
+		if (
+			!availableLocalesArray.some((locale) => locale.id === defaultLocale)
+		) {
+			return availableLocalesArray;
+		}
+
```
Comment (dsanz) [inline modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js:49]: Some thoughts:

- I'm curious about why we worry about `defaultLocale` but not about other potential things that can go wrong here, like the `availablelocales` or the `locales`. I'd recommend a more general approach if the goal is to harden component logic. 
- There's more logic which makes use of the `defaultLocale` in the component. If the value of defaultLocale is wrong, that logic may break too (see [setEditingLocale()](https://github.com/liferay/liferay-portal/blob/21078d8663ae2a8d264f611a68d0e17c4ef9a056/modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/index.js#L70) invocation for example). So, to harden the logic we would have to deal with a "reasonable" value for this even if it comes `undefined` or contains a wrong value.
- As this component is managed via tag, we'd probably prefer to [control the parameters when being provided](https://github.com/liferay/liferay-portal/blob/991e5c1030b6382d07fb1bf4a5ef9fdb3ebc329c/modules/apps/frontend-taglib/frontend-taglib/src/main/java/com/liferay/frontend/taglib/servlet/taglib/TranslationManagerTag.java#L183-L191) to the component instead. This way we have access to server-side info like the available locales and can make corrections easier. We could do that frontend side too as there are APIs to get the languages, however, that may not be the case for other parameters.

So I'd suggest to approach this from the tag instead. What do you think @tinycarol @markocikos ?

Comment (tototrinh) [reply modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js:49]: Hi @dsanz
Sorry for not taking a more general approach, I just focused on the `defaultLocale` mentioned in the ticket.

> we'd probably prefer to [control the parameters when being provided](https://github.com/liferay/liferay-portal/blob/991e5c1030b6382d07fb1bf4a5ef9fdb3ebc329c/modules/apps/frontend-taglib/frontend-taglib/src/main/java/com/liferay/frontend/taglib/servlet/taglib/TranslationManagerTag.java#L183-L191) to the component instead

It seems only DDMForm is currently using the taglib `liferay-frontend:translation-manager` and the change on https://github.com/liferay-frontend/liferay-portal/pull/2761/commits/63663562fc4cc4bb1857c58690987e49b62845b4 makes sure that the DefaultLanguageId is valid on the server side (will get a valid value [here](https://github.com/liferay/liferay-portal/blob/991e5c1030b6382d07fb1bf4a5ef9fdb3ebc329c/modules/apps/frontend-taglib/frontend-taglib/src/main/java/com/liferay/frontend/taglib/servlet/taglib/TranslationManagerTag.java#L186)). Do you think that's enough for this issue only with https://github.com/liferay-frontend/liferay-portal/pull/2761/commits/63663562fc4cc4bb1857c58690987e49b62845b4?

Comment (markocikos) [reply modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js:49]: It looks like it's a valid case that `defaultLanguageId` is not set. It's not a required tag attribute, not default in tag, no default in the React component. I think it's fine to add a check for it on frontend side. The code in this PR even checks for the case where default is different from available set, which is nice, covering another fringe case :+1: 

In general, when it comes to code improvement that add robustness, it is a balancing act. It's always possible to add more checks, to the point of unnecessarily bloating the code. I think it's fine to merge improvements only in the scope of the ticket.

**Review by markocikos — APPROVED** [review]

File: modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js (hunk 1 above)
Comment (dsanz) [reply modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/components/LocalesList.js:49]: That's valid concerns Marko, we have to validate defaults. I'd say _setting_ a default language must be covered anyways as it's existence is taken for granted in DXP. So if any invocation of the taglib/react component does not have it, that should be considered a wrong usage, so it's fine if the component can handle that gracefully (I'd expect anyways some error in the console to signal such a bad usage). I still see a bunch of other cases not covered in the code so far, and generally we don't cover them holistically, as you said, that would bloat the code. 

Just a note to clarify: defaultLanguage in the LPS ticket is not undefined, but altered to have an invalid value. That means that the array entry index is wrong not only when sorting the locales but in any case, unless the DDM use case. That's not wrong per se, but it feels incomplete from a component perspective.

Let's go ahead with this one anyways as it adds value for a case

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**